core/src/preferences/PreferencesBackend.cpp

```diff
@@ -289,6 +325,19 @@ namespace ledger {
             _db = obtainInstance(_dbName);
         }
 
+        std::string PreferencesBackend::getEncryptionSalt() {
+            auto saltKey = std::vector<uint8_t>(ENCRYPTION_SALT_KEY.cbegin(), ENCRYPTION_SALT_KEY.cend());
+
+            // move _cipher around to prevent trying to decrypt the salt
+            auto cipher = std::move(_cipher);
+            _cipher = Option<AESCipher>();
+
+            auto salt = get(saltKey).value_or("");
```

Comment:
So maybe use raw function instead of get() , to not dancing with _cipher ?

Reply:
:+1: